modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java

```diff
@@ -107,9 +122,32 @@ protected AbstractMavenDeployableProjectModelBuilder(MavenConfiguration mavenCon
   public final DeployableProjectModel build() {
     File pom = getPomFromFolder(projectFolder);
 
+    Properties pomProperties;
+    Optional<String> version = empty();
+
+    if (projectFolder.equals(APP) || projectFolder.equals(DOMAIN) || projectFolder.isDirectory()) {
+      try {
+        // 1) look for pom.properties, get the version info from there
+        pomProperties = getPomPropertiesFolder(projectFolder);
+        version = ofNullable(pomProperties.getProperty("version"));
+      } catch (ArtifactDescriptorCreateException e) {
+        logger.debug("unable to get version info from pom.properties:" + e.getMessage());
+      }
+    }
+
     List<String> activeProfiles = mavenConfiguration.getActiveProfiles().orElse(emptyList());
     MavenPomParser parser = MavenPomParserProvider.discoverProvider().createMavenPomParserClient(pom.toPath(), activeProfiles);
 
+    // 2) if a version is passed using system properties, use the version instead
+    String originalPomVersion = parser.getModel().getVersion();
+    Pattern pattern = compile("\\$\\{\\s*([^}]*)\\s*\\}");
+    Matcher matcher = pattern.matcher(originalPomVersion);
+    if (matcher.find()) {
+      String potentialProperty = matcher.group(1);
+      if (getProperty(potentialProperty) != null) {
+        version = ofNullable(originalPomVersion.replaceAll("\\$\\{[^}]*\\}", getProperty(potentialProperty)));
```

Comment:
use the capturing group from the previos regex instead of creating a new one for replaceAll